src/lib/utils/github/github.ts

```diff
@@ -1,55 +1,58 @@
-import octokit from './octokit';
+import type { Octokit } from '@octokit/rest';
 import { Buffer } from 'buffer';
 
-export async function getRepoByOwnerAndName(owner: string, repo: string) {
-  const { data } = await octokit.request('GET /repos/{owner}/{repo}', {
-    owner,
-    repo,
-  });
+export default class GitHub {
+  private octokit: Octokit;
 
-  return data;
-}
-
-export async function getRepoByUrl(repoUrl: string) {
-  const url = new URL(repoUrl);
-
-  if (url.host !== 'github.com') {
-    throw new Error(`Invalid host: ${url.host}`);
+  constructor(octokit: Octokit) {
```

Comment:
thought about it but i kinda like enforcing passing the client explicitly, bc otherwise you might accidentally make unauthorized calls that have a very strict rate limit 😬